### PR TITLE
feat: added hooks for pyinstaller

### PR DESCRIPTION
**Pull request by kumaraditya303**
For review instructions:

- Create a file `main.py`
- add this to the file 
```python
import os

from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    page.goto("http://whatsmyuseragent.org/")
    page.screenshot(path="example.png")
    browser.close()

```

- If you want to bundle browsers then set `PLAYWRIGHT_BROWSERS_PATH=0` to include bowser's inside the binary and then download the browsers 
- installer pyinstaller 
```sh
pip install pyinstaller
```

- Run pyinstaller on the file like 
```sh
pyinstaller -F main.py
```

- Binary would be placed in `dist/` relative to current dir and test it

### CODE REVIEW

**Comment by mxschmitt** [comment]
10+ individual users requested it, so makes sense to add support for it.

Fixes #513

**Review by mxschmitt — APPROVED** [review]
LGTM, tested with the following test case:
1. `PLAYWRIGHT_BROWSERS_PATH=0 playwright install chromium`
2. `pyinstaller -F test.py`
3. clean browsers to make sure we are not using the wrong ones
4. `PLAYWRIGHT_BROWSERS_PATH=0 ./dist/test`

**Comment by kumaraditya303** [comment]
cc: @pavelfeldman

**Comment by mxschmitt** [comment]
We can land if you somehow can set the env var for the driver automatically. Like detect somehow if its pyinstaller and then set it.

**Comment by kumaraditya303** [comment]
> We can land if you somehow can set the env var for the driver automatically. Like detect somehow if its pyinstaller and then set it.
> 
> Had an offline discussion with him about it.

Done!

File: playwright/_impl/_transport.py [hunk 1]
```diff
@@ -105,6 +105,10 @@ async def run(self) -> None:
             creationflags = subprocess.CREATE_NO_WINDOW
 
         try:
+            env = os.environ.copy()
+            if getattr(sys, "frozen", False):
```
Comment (mxschmitt) [inline playwright/_impl/_transport.py:109]: nit: comment that its for Pyinstaller.

**Review by mxschmitt — APPROVED** [review]
LGTM

File: playwright/_impl/_transport.py [hunk 2]
```diff
@@ -105,6 +105,10 @@ async def run(self) -> None:
             creationflags = subprocess.CREATE_NO_WINDOW
 
         try:
+            env = os.environ.copy()
```
Comment (kumaraditya303) [inline playwright/_impl/_transport.py:108]:
```suggestion
            # For pyinstaller
            env = os.environ.copy()
```

**Pull request merged**